infra/build_specified_commit.py

```diff
@@ -68,16 +68,17 @@ def detect_main_repo(project_name, repo_name=None, commit=None):
   """
 
   if not repo_name and not commit:
-    print('Error: can not detect main repo without a repo_name or a commit.')
+    logging.error(
+        'Error: can not detect main repo without a repo_name or a commit.')
```

Comment:
nit: I think `Error: ` is redundant here, `logging.error` will make it explicit that it's an error